java/vector/src/test/java/org/apache/arrow/vector/file/BaseFileTest.java

```diff
@@ -408,28 +410,66 @@ protected void validateNestedDictionary(VectorSchemaRoot root, DictionaryProvide
+  protected VectorSchemaRoot writeDecimalData(BufferAllocator bufferAllocator) {
+    NullableDecimalVector decimalVector1 = new NullableDecimalVector("decimal1", bufferAllocator, 10, 3);
+    NullableDecimalVector decimalVector2 = new NullableDecimalVector("decimal2", bufferAllocator, 4, 2);
+    NullableDecimalVector decimalVector3 = new NullableDecimalVector("decimal3", bufferAllocator, 16, 8);
 
-    ListVector.Accessor accessor = vector.getAccessor();
-    Assert.assertEquals(3, accessor.getValueCount());
-    Assert.assertEquals(Arrays.asList(0, 1), accessor.getObject(0));
-    Assert.assertEquals(Arrays.asList(0), accessor.getObject(1));
-    Assert.assertEquals(Arrays.asList(1), accessor.getObject(2));
+    int count = 10;
+    decimalVector1.allocateNew(count);
+    decimalVector2.allocateNew(count);
+    decimalVector3.allocateNew(count);
 
-    Dictionary dictionary = provider.lookup(2L);
-    Assert.assertNotNull(dictionary);
-    NullableVarCharVector.Accessor dictionaryAccessor = ((NullableVarCharVector) dictionary.getVector()).getAccessor();
-    Assert.assertEquals(2, dictionaryAccessor.getValueCount());
-    Assert.assertEquals(new Text("foo"), dictionaryAccessor.getObject(0));
-    Assert.assertEquals(new Text("bar"), dictionaryAccessor.getObject(1));
+    for (int i = 0; i < count; i++) {
+      writeIntAsDecimal(decimalVector1, i, i);
+      writeIntAsDecimal(decimalVector2, i * (1 << 10), i);
+      //long blah = i * Long.valueOf(String.format("%0" + 10 + "d", 0).replace("0", "1"));
+      //System.out.println("*** Int val: " + blah); (Long.valueOf(String.format("%0" + 16 + "d", 0).replace("0", "1"))), i);
+      writeIntAsDecimal(decimalVector3, i * 1111111111111111L, i);
+    }
+
+    decimalVector1.getMutator().setValueCount(count);
+    decimalVector2.getMutator().setValueCount(count);
+    decimalVector3.getMutator().setValueCount(count);
+
+    List<Field> fields = ImmutableList.of(decimalVector1.getField(), decimalVector2.getField(), decimalVector3.getField());
+    List<FieldVector> vectors = ImmutableList.<FieldVector>of(decimalVector1, decimalVector2, decimalVector3);
+    return new VectorSchemaRoot(fields, vectors, count);
+  }
+
+  protected void validateDecimalData(VectorSchemaRoot root) {
+    NullableDecimalVector decimalVector1 = (NullableDecimalVector) root.getVector("decimal1");
+    NullableDecimalVector decimalVector2 = (NullableDecimalVector) root.getVector("decimal2");
+    NullableDecimalVector decimalVector3 = (NullableDecimalVector) root.getVector("decimal3");
+    int count = 10;
+    Assert.assertEquals(count, root.getRowCount());
+
+    for (int i = 0; i < count; i++) {
+      // Verify decimal 1 vector
+      BigDecimal readValue = decimalVector1.getAccessor().getObject(i);
+      ArrowType.Decimal type = (ArrowType.Decimal) decimalVector1.getField().getType();
+      BigDecimal genValue = new BigDecimal(BigInteger.valueOf(i), type.getScale());
```

Comment:
it would be much cleaner to get the scale with #972 merged

Reply:
@BryanCutler is #972 just for testing purposes?

Reply:
No, that PR came from me adding Timestamp to Spark. I needed to check the time zone string given a NullableTimeStampMicroTZVector

Reply:
I see.